doc: [WD-31419] Add Local Peering Documentation (Networks) (#17217)
canonical/lxd: doc/howto/network_ovn_peers.md

When creating the peer routing relationship, specify a peering name that identifies the relationship for the respective network.
The name can be chosen freely, and you can use it later to edit or delete the relationship.

> ```{important}

Author: I just moved the pre-existing note to a different place, but I'll make this amendment.

> If the project or the network name is incorrect, the command will not return any error indicating that the respective project/network does not exist, and the routing relationship will remain in pending state.

Reviewer: ```suggestion
If the project or the network name is incorrect, the command does not return any error indicating that the respective project/network does not exist, and the routing relationship remains in pending state.
```
For brevity and consistency, prefer present tense unless switching tense is necessary.

    lxc network peer create <network1> <peering_name> <project2/network2> [configuration_options] --project=<project1>
    lxc network peer create <network2> <peering_name> <project1/network1> [configuration_options] --project=<project2>

````
````{group-tab} UI

From the {guilabel}`Networks` page of the {ref}`web UI <access-ui>`, select the desired OVN network. On the network's {guilabel}`Local Peerings` tab, click {guilabel}`Create local peering`.

Fill in all required fields in the {guilabel}`Create local peering` panel.

```{figure} /images/networks/network_create_local_peerings.png
:width: 60%
:alt: View a list of local peerings on a network
```

> Target projects and networks for which you have read permission are available from the dropdown selectors. If you want to use a project or network not available in the dropdown, choose the {guilabel}`Manually enter` option

Reviewer: ```suggestion
Target projects and networks for which you have read permission are available from the dropdown selectors. If you want to use a project or network not available in the dropdown, choose the {guilabel}`Manually enter` option.
```

    lxc network peer create <network1> <peering_name> <project2/network2> [configuration_options] --project=<project1>
    lxc network peer create <network2> <peering_name> <project1/network1> [configuration_options] --project=<project2>

````
````{group-tab} UI

> In {ref}`the web UI <access-ui>`, select {guilabel}`Networks` in the left sidebar, then select the desired OVN network. On the resulting screen, view the {guilabel}`Local Peerings` tab. Click the {guilabel}`Create local peering` button.

Reviewer: ```suggestion
From the {guilabel}`Networks` page of the {ref}`web UI <access-ui>`, select the desired OVN network. On the network's {guilabel}`Local Peerings` tab, click {guilabel}`Create local peering`.
```

> In {ref}`the web UI <access-ui>`, select {guilabel}`Networks` in the left sidebar, then select the desired OVN network. On the resulting screen, view the {guilabel}`Local Peerings` tab. Click the {guilabel}`Create local peering` button.

Reviewer: Question to @edlerd : Should this also be changed in the UI then? The tab label is "Local Peerings".

There's also "Create Local peering" for a modal title — should this be "Create local peering"?

> In {ref}`the web UI <access-ui>`, select {guilabel}`Networks` in the left sidebar, then select the desired OVN network. On the resulting screen, view the {guilabel}`Local Peerings` tab. Click the {guilabel}`Create local peering` button.

Reviewer 2: Yes, I think we should adjust it in the UI as well.

> In the {guilabel}`Create Local Peering` panel, the {guilabel}`Name`, {guilabel}`Target project` and {guilabel}`Target Network` fields are required.

Reviewer: ```suggestion
Fill in all required fields in the {guilabel}`Create local peering` panel.
```

```{figure} /images/networks/network_create_local_peerings.png
:width: 60%
:alt: View a list of local peerings on a network
```

> - For the {guilabel}`Target project`, you may provide the name of a project that you do not have edit permission for by clicking {guilabel}`Manually enter project` within the dropdown selector. This can also be done for the target network.

Reviewer: ```suggestion
- If you want to use a project for which you do not have read permission for the {guilabel}`Target project`, select {guilabel}`Manually enter project` from the dropdown selector. You can also do this for the {guilabel}`Target network`.
```

> You can optionally create a mutual peering between two networks that you have edit permissions for, by clicking the {guilabel}`Create mutual peering` checkbox.

Author: Note: Discussed in private conversation that "Edit" permissions is fine.

> Once you have finished setting up the local peering, click the {guilabel}`Create local peering` button.

Reviewer: ```suggestion
```

Self-explanatory.

## List routing relationships

`````{tabs}
````{group-tab} CLI

To list all network peerings for a network, use the following command:

    lxc network peer list <network>

````
````{group-tab} UI

In {ref}`the web UI <access-ui>`, select {guilabel}`Networks` in the left sidebar, then select the desired OVN network. On the resulting screen, view the {guilabel}`Local Peerings` tab:

```{figure} /images/networks/network_list_local_peerings.png
:width: 95%
:alt: View a list of local peerings on a network
```
````
`````

> 

Reviewer: ```suggestion
```

## Edit a routing relationship

`````{tabs}
````{group-tab} CLI

Use the following command to edit a network peering:

    lxc network peer edit <network> <peering_name>

This command opens the network peering in YAML format for editing.

````
````{group-tab} UI

> In {ref}`the web UI <access-ui>`, select {guilabel}`Networks` in the left sidebar, then select the desired OVN network. On the resulting screen, view the {guilabel}`Local Peerings` tab. This tab shows you information about all local peerings on the network. Click the {guilabel}`Edit` icon next to a local peering to edit it:

Reviewer: ```suggestion
In {ref}`the web UI <access-ui>`, select {guilabel}`Networks` in the left sidebar, then select the desired OVN network. On the resulting screen, view the {guilabel}`Local peerings` tab. This tab shows you information about all local peerings on the network. Click the {guilabel}`Edit` icon next to a local peering to edit it:
```
